crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -254,6 +254,19 @@ def prepare(options = {})
 
       protected
 
+      def finalize_cluster_upgrade(cluster)
+        save_upgrade_state("Finalizing the upgrade of nodes in cluster #{cluster}")
+
+        NodeObject.find("pacemaker_config_environment:#{cluster}").each do |node|
+          if node["pacemaker"]["founder"] && node.roles.include?("neutron-network")
+            node.run_ssh_cmd("source /root/.openrc; neutron-ha-tool --l3-agent-rebalance")
```

Comment:
@rhafer I've added a task to 
https://trello.com/c/MquyXrTn/2058-8-s49p16-orchestrate-upgrade-of-controller-nodes-in-ha-setup-upgrade-step-12-part-2-ha-multiple-clusters